spec/requests/api/v3/providers/courses/show_spec.rb

```diff
@@ -0,0 +1,63 @@
+require "rails_helper"
+
+describe "GET v3/providers/:provider_code/courses/:course_code" do
+  let(:recruitment_cycle) { find_or_create :recruitment_cycle }
+  let(:provider) { create :provider, recruitment_cycle: recruitment_cycle }
+  let(:jsonapi_course) {
+    JSON.parse(
+      JSONAPI::Serializable::Renderer.new.render(
+        course,
+        class: {
+          Course: API::V2::SerializableCourse,
+        },
+      ).to_json,
+    )
+  }
+  let(:jsonapi_response) { JSON.parse(response.body) }
+  let(:route) {
+    "/api/v3/recruitment_cycles/#{recruitment_cycle.year}" \
+    "/providers/#{provider.provider_code}" \
+    "/courses/#{course.course_code}"
+  }
+
+  context "with a published course" do
+    let(:enrichment) { build :course_enrichment, :published }
+    let(:course) {
+      create(:course,
+             provider: provider,
+             enrichments: [enrichment])
+    }
+
+    it "returns full course information" do
+      get route
+
+      expect(jsonapi_response["data"]).to eq jsonapi_course["data"]
+    end
+
+    it "returns sparse course information" do
+      requested_fields = %w[course_code name provider_code].sort
+      get route + "?fields[courses]=#{requested_fields.join(',')}"
+
+      expect(jsonapi_response["data"]["attributes"].keys).to eq requested_fields
+    end
+  end
```

Comment:
maybe add a test:
- for a draft course enrichment
- the draft data is not returned

Reply:
Done